src/PureCloud.Client/Apis/SearchApi.cs

```diff
@@ -222,4 +222,317 @@ public async Task<SitesSearchResponse> CreateTelephonyProvidersEdgesSitesSearchA
 
         return await response.Content.ReadFromJsonAsync<SitesSearchResponse>(_options.JsonSerializerOptions, cancellationToken);
     }
+
+    /// <inheritdoc />
+
+    public async Task<GKNDocumentationSearchResponse> GetDocumentationGknSearchAsync(string q64, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(q64);
+
+        var parameters = new NameValueCollection { { "q64", q64 } };
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = UriHelper.GetUri("api/v2/documentation/gkn/search", parameters);
+
+        var response = await client.GetAsync(uri, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<GKNDocumentationSearchResponse>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc />
+
+    public async Task<DocumentationSearchResponse> GetDocumentationSearchAsync(string q64, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(q64);
+
+        var parameters = new NameValueCollection { { "q64", q64 } };
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = UriHelper.GetUri("api/v2/documentation/search", parameters);
+
+        var response = await client.GetAsync(uri, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<DocumentationSearchResponse>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc />
+
+    public async Task<GroupsSearchResponse> GetGroupsSearchAsync(string q64, IEnumerable<string> expand = null, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(q64);
+
+        var parameters = new NameValueCollection { { "q64", q64 } };
+
+        if (expand != null)
+        {
+            foreach (var item in expand)
+            {
+                parameters.Add("expand", item);
+            }
+        }
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = UriHelper.GetUri("api/v2/groups/search", parameters);
+
+        var response = await client.GetAsync(uri, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<GroupsSearchResponse>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc />
+
+    public async Task<LocationsSearchResponse> GetLocationsSearchAsync(string q64, IEnumerable<string> expand = null, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(q64);
+
+        var parameters = new NameValueCollection { { "q64", q64 } };
+
+        if (expand != null)
+        {
+            foreach (var item in expand)
```

Comment:
Same as above. When the argument is plural, the name should be in plural form and the iteration variable is single form